src/bun.js/bindings/KeyObject.cpp

```diff
@@ -1239,6 +1246,492 @@ JSC::EncodedJSValue KeyObject__createSecretKey(JSC::JSGlobalObject* lexicalGloba
+                } else if (encoding_str == "der"_s) {
+                    params.encoding = CryptoAlgorithmECDSAEncoding::DER;
+                } else {
+                    JSC::throwTypeError(globalObject, scope, "invalid dsaEncoding"_s);
+                    return JSC::JSValue::encode(JSC::JSValue {});
+                }
+            }
+        }
+        auto result = WebCore::CryptoAlgorithmECDSA::platformSign(params, ec, vectorData);
+        if (result.hasException()) {
+            WebCore::propagateException(*globalObject, scope, result.releaseException());
+            return JSC::JSValue::encode(JSC::JSValue {});
+        }
+        auto resultData = result.releaseReturnValue();
+        auto size = resultData.size();
+        auto* buffer = jsCast<JSUint8Array*>(JSValue::decode(JSBuffer__bufferFromLength(globalObject, size)));
+        if (size > 0)
+            memcpy(buffer->vector(), resultData.data(), size);
+
+        return JSC::JSValue::encode(buffer);
+    }
+    case CryptoKeyClass::RSA: {
+        const auto& rsa = downcast<WebCore::CryptoKeyRSA>(wrapped);
+        CryptoAlgorithmIdentifier restrict_hash;
+        bool isRestrictedToHash = rsa.isRestrictedToHash(restrict_hash);
+        if (isRestrictedToHash && hash != restrict_hash) {
+            JSC::throwTypeError(globalObject, scope, "digest not allowed"_s);
+            return JSC::JSValue::encode(JSC::JSValue {});
+        }
+        switch (rsa.algorithmIdentifier()) {
+        case CryptoAlgorithmIdentifier::RSASSA_PKCS1_v1_5: {
+            auto result = (customHash) ? WebCore::CryptoAlgorithmRSASSA_PKCS1_v1_5::platformSignWithAlgorithm(rsa, hash, vectorData) : CryptoAlgorithmRSASSA_PKCS1_v1_5::platformSign(rsa, vectorData);
+            if (result.hasException()) {
+                WebCore::propagateException(*globalObject, scope, result.releaseException());
+                return JSC::JSValue::encode(JSC::JSValue {});
+            }
+            auto resultData = result.releaseReturnValue();
+            auto size = resultData.size();
+            auto* buffer = jsCast<JSUint8Array*>(JSValue::decode(JSBuffer__bufferFromLength(globalObject, size)));
+            if (size > 0)
+                memcpy(buffer->vector(), resultData.data(), size);
+
+            return JSC::JSValue::encode(buffer);
+        }
+        case CryptoAlgorithmIdentifier::RSA_PSS: {
+            CryptoAlgorithmRsaPssParams params;
+            params.padding = RSA_PKCS1_PADDING;
+            if (count > 4) {
+                auto padding = callFrame->argument(4);
+                if (!padding.isUndefinedOrNull() && !padding.isEmpty()) {
+                    if (!padding.isNumber()) {
+                        JSC::throwTypeError(globalObject, scope, "padding is expected to be a number"_s);
+                        return JSC::JSValue::encode(JSC::JSValue {});
+                    }
+                    params.padding = padding.toUInt32(globalObject);
```

Comment:
This will throw if `toUint32` is negative I think?

```suggestion
                    params.padding = padding.toUInt32(globalObject);
                    RETURN_IF_EXCEPTION(throwScope, {});
```